documentation/Software Requirements Specification.md

```diff
@@ -0,0 +1,234 @@
+<h2 align="center">Managing departments and employees</h2>
+
+<!--TABLE OF CONTENTS-->
+<details open="open">
+  <summary>Table of Contents</summary>
+  <ol>
+    <li>
+      <a href="#vision">Vision</a>
+    </li>
+    <li>
+      <a href="#department">Department</a>
+      <ul>
+        <li><a href="#display-list-of-department">Display list of department</a></li>
+        <li><a href="#add-department">Add department</a></li>
+        <li><a href="#edit-department">Edit department</a></li>
+        <li><a href="#removing-the-department">Removing the department</a></li>
+      </ul>
+    </li>
+    <li>
+        <a href="#employee">Employee</a>
+        <ul>
+        <li><a href="#display-list-of-employee">Display list of employee</a></li>
+        <li><a href="#add-employee">Add employee</a></li>
+        <li><a href="#edit-employee">Edit employee</a></li>
+        <li><a href="#removing-the-employee">Removing the employee</a></li>
+        </ul>
+    </li>
+  </ol>
+</details>
+<!--/TABLE OF CONTENTS-->
+
+<!--VISION-->
+## Vision
+
+"Department and Employee Management" is a web application that allows an administrator to record information about a department and an employee.
+
+Application should provide:
+
+* Storing list of department and list of employee in a database;
+* Display list of department and the average salary for these departments;
+* Updating the list of department (adding, editing, removing);
+* Display list of employee in the departments with an indication of the salary for each employee;
+* Updating the list of employee (adding, editing, removing);
+* Filtering by employees born on a certain date or in the period between dates.
+<!--/VISION-->
+
+<!--DEPARTMENT-->
+## Department
+### Display list of department
+
+The mode is designed to view the list of department.
+
+<b>Main scenario:</b>
+* User selects item “Department”;
+* Application displays list of Department.
+
+![](department_listt.jpg)
+Pic. 2.1 View the Department list.
+
+The list displays the following columns:
+* ID – unique department number;
+* Nubmer of employees - the number of employees in this department
+* The average salary ($) - average salary of all employees of the department
+### Add department
+
+<b>Main scenario:</b>
+* Administrator clicks the “Add” button in the department list view mode;
+* Application displays form to enter department data;
+* Administrator enters department data and presses “Save” button;
+* If any data is entered incorrectly, incorrect data messages are displayed;
+* If entered data is valid, then record is adding to database;
+* If error occurs, then error message is displaying;
```

Comment:
It would be great to see a mockup with error messages.

Reply:
Fixed